pkg/scheduler/framework/types.go

```diff
@@ -0,0 +1,99 @@
+/*
```

Comment:
I am afraid that it might be the same for many utilities in this file as well. The extension point can calculate them directly rather than wrapping the same set of data in a different struct.